node/src/components/storage.rs

```diff
@@ -1689,7 +1697,8 @@ fn garbage_collect_block_body_v2_db(
                         })
                     }
                 };
-            live_digests.insert(key_to_part_db);
+            live_digests[live_digests_index].insert(key_to_part_db);
```

Comment:
So if we had a block in storage where the last  `merkle_proof_of_rest` wasn't `SENTINEL1` we would panic here?
E.g. if we had added a field to the block and than somehow started an earlier version of the node software?

Reply:
Yes, that's probably what would happen. I'm not sure what we _should_ be doing here, though.

Reply:
Not sure either; I'm fine with it. Let's wait for a second reviewer's opinion.

Reply:
I suggest we use a `Map<usize, Vec<Digest>>` for `live_digest`s. That way we can gracefully handle extending block bodies as needed.

Reply:
Hmm, so that if an index doesn't exist, we just extend the map here? I guess that would work - then we'd just not check the entries that correspond to fields that were added later. Maybe we could even print some kind of a warning if such entries exist?

Reply:
So having talked it over, the solution you and I settled on was to create a `BlockBody::PARTS_COUNT` constant and use that to bound `live_digest`s.  See https://github.com/casper-network/casper-node/pull/1708/commits/fd2b362e86d906a2159e9d2595ab3cfc5614450f